src/async-components/views/dialogs/security/CreateKeyBackupDialog.tsx

```diff
@@ -75,24 +74,35 @@ export default class CreateKeyBackupDialog extends React.PureComponent<IProps, I
         this.setState({
             error: undefined,
         });
-        let info: IKeyBackupInfo | undefined;
         const cli = MatrixClientPeg.safeGet();
         try {
-            await accessSecretStorage(async (): Promise<void> => {
-                // `accessSecretStorage` will have bootstrapped secret storage if necessary, so we can now
-                // set up key backup.
-                //
-                // XXX: `bootstrapSecretStorage` also sets up key backup as a side effect, so there is a 90% chance
-                // this is actually redundant.
-                //
-                // The only time it would *not* be redundant would be if, for some reason, we had working 4S but no
-                // working key backup. (For example, if the user clicked "Delete Backup".)
-                info = await cli.prepareKeyBackupVersion(null /* random key */, {
-                    secureSecretStorage: true,
+            // Check if 4S already set up
+            const secretStorageAlreadySetup = await cli.hasSecretStorageKey();
+
+            if (!secretStorageAlreadySetup) {
+                // bootstrap secret storage, that will also create a backup version
+                await accessSecretStorage(async (): Promise<void> => {
+                    // do nothing, all is now setup correctly
                 });
-                info = await cli.createKeyBackupVersion(info);
-            });
-            await cli.scheduleAllGroupSessionsForBackup();
+            } else {
+                // Secret storage exists, we need to ensure that we can write to it before
+                // we create a new backup version. It ensures that we can write to it and keep it in sync.
+                await withSecretStorageKeyCache(async () => {
+                    const crypto = cli.getCrypto();
+                    if (!crypto) {
+                        throw new Error("End-to-end encryption is disabled - unable to create backup.");
+                    }
+
+                    // this is the secret that will need to be updated, ensure that we can access it.
+                    // This will ask the user to enter their passphrase/key.
+                    await cli.secretStorage.get("m.megolm_backup.v1");
+                    // if we get here, we can access the secret storage, so we can create a backup version.
+                    // We don't reset if we can't access the secret storage, as the secret storage will then
+                    // have an outdated secret for backup that future sessions will not be able to use.
+                    await crypto.resetKeyBackup();
```

Comment:
Minor suggestions for clarity

```suggestion
                    // Before we reset the backup, let's make sure we can access secret storage, to
                    // reduce the chance of us getting into a broken state where we have an outdated
                    // secret in secret storage.
                    // `SecretStorage.get` will ask the user to enter their passphrase/key if necessary;
                    // it will then be cached for the actual backup reset operation.
                    await cli.secretStorage.get("m.megolm_backup.v1");

                    // We now know we can store the new backup key in secret storage, so it is safe to
                    // go ahead with the reset.
                    await crypto.resetKeyBackup();
```